sscanss/editor/json_attributes.py

```diff
@@ -0,0 +1,858 @@
+        :rtype: QComboBox
+        """
+        self.combo_box = QtWidgets.QComboBox()
+        self.combo_box.addItems(self.list_reference.getObjectKeys())
+        self.combo_box.setCurrentText(str(self.value))
+        self.combo_box.currentIndexChanged.connect(self.selectNewIndex)
+
+        return self.combo_box
+
+
+class OrderItem:
+    """The container for the items in the ObjectOrder
+    :param text: the text of the item
+    :type text: str
+    :param included: whether the item should be included in the json
+    :type included: bool
+    """
+    def __init__(self, text, included=True):
+        self.text = text
+        self.included = included
+
+    def __eq__(self, other):
+        return self.text == other.text and self.included == other.included
+
+
+class ObjectOrder(ListReference):
+    """Attribute contains a custom order of objects in referenced list"""
+    default_value = []
+
+    def __init__(self, list_path, include_all=True, initial_value=None):
+        super().__init__(list_path, initial_value)
+        self.include_all = include_all
+        self.json_set = False
+
+    def getWidgetItems(self):
+        return [self.order_list.item(i).text() for i in range(self.order_list.count())]
+
+    def findItemIndex(self, name):
+        for index, item in enumerate(self.value):
+            if item.text == name:
+                return index
+
+    def itemDropped(self):
+        """Should be called when an item is dragged and dropped to update the current value by getting all values
+        from list widget"""
+        new_value = [self.value[self.findItemIndex(name)] for name in self.getWidgetItems()]
+        self.setValue(new_value)
+        self.updateUi()
+
+    def itemDoubleClicked(self, clicked_item):
+        """Should be called when item in the list is clicked and select or diselect it
+        :param clicked_item: the item from QListWidget which was clicked
+        :type clicked_item: QListWidgetItem
+        """
+        if not self.include_all:
+            item_index = self.findItemIndex(clicked_item.text())
+            self.value[item_index].included = not self.value[item_index].included
+            self.been_set.emit(self.value[item_index].included)
+            self.updateUi()
+
+    def updateOnListChange(self):
+        """Should be called on every time the referenced list is updated. It first adds all the objects
+        which were previously included in their previous order and then adds all the new objects
+        """
+        if self.json_set:
+            for item in self.value:
+                if item.text not in self.list_reference.getObjectKeys():
+                    return
+
+        new_items = []
+        for item in self.value:
+            if item.text in self.list_reference.getObjectKeys():
+                new_items.append(item)
+
+        for item in self.list_reference.getObjectKeys():
+            if item not in [item.text for item in self.value]:
```

Comment:
Sorry I missed this. 
This list is created multiple times in a loop **[item.text for item in self.value]**
Please fix